src/observable/ScalarObservable.ts

```diff
@@ -80,8 +80,8 @@ proto.reduce = function <T, R>(project: (acc: R, x: T) => R, seed?: R): Observab
     return <any>this;
   }
   let result = tryCatch(project)(seed, this.value);
-  if (result === errorObject) {
-    return new ErrorObservable(errorObject.e);
+  if (result as any === errorObject) {
```

Comment:
> I'll add in the errorObject type change, it is probably small enough that it shouldn't be a big deal.

: do you mean bringing change https://github.com/ReactiveX/RxJS/pull/1077#discussion_r47718354 mentioned in here?